src/xalt_initialize.c

```diff
@@ -503,14 +503,24 @@ void myinit(int argc, char **argv)
   if (v == NULL)
     v = XALT_GPU_TRACKING;
   xalt_gpu_tracking = (strcmp(v,"yes") == 0);
-  
+
   do
     {
       if (xalt_gpu_tracking)
         {
           DEBUG0(stderr, "  GPU tracing\n");
 
 #ifdef USE_NVML
+          /* check whether the nvidia module is loaded */
+          char nvidiaD[] =  "/sys/module/nvidia";
+          struct stat s = {0};
+
+          if ( xalt_gpu_tracking && (stat(nvidiaD, &s) !=  0 || ! S_ISDIR(s.st_mode))) {
```

Comment:
nit: this is already inside the `if (xalt_gpu_tracking)` branch, so it's probably unnecessary to check xalt_gpu_tracking again here.